OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -82,6 +84,37 @@ public class MobileInfo : ConditionalTraitInfo, IMoveInfo, IPositionableInfo, IO
 		[Desc("The condition to grant to self while inside a tunnel.")]
 		public readonly string TunnelCondition = null;
 
+		[Desc("Can this unit move underground?")]
+		public readonly bool Subterranean = false;
+
+		[GrantedConditionReference]
+		[Desc("The condition to grant to self while underground.")]
+		public readonly string SubterraneanCondition = null;
+
+		[Desc("Pathfinding cost for submerging or reemerging")]
```

Comment:
Minor style nit: It would look better/be more consistent if you added a full stop at the end.